pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/creator/impl/SegmentIndexCreationDriverImpl.java

```diff
@@ -374,6 +374,7 @@ public static void persistCreationMeta(File indexDir, long crc, long creationTim
   void buildIndexCreationInfo()
       throws Exception {
     Set<String> varLengthDictionaryColumns = new HashSet<>(_config.getVarLengthDictionaryColumns());
+    Set<String> rawIndexCreationColumns = _config.getRawIndexCreationColumns();
```

Comment:
Good catch. We use 2 configs for no-dictionary column, so we might want to check `_config.getRawIndexCompressionType()` as well (the `createDictionary` flag is not checked though, but +1 on set it correctly)

Reply:
Yes, "`createDictionary` flag is not checked", and instead `getColumnCompressionType(segmentCreationSpec, fieldSpec)` is called to check for dict columns.